src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -2000,6 +2002,48 @@ class spell_dk_ghoul_thrash : public SpellScriptLoader
     }
 };
 
+enum DancingRuneWeapon
+{
+    NPC_DANCING_RUNE_WEAPON = 27893,
+
+    DATA_INITIAL_TARGET_GUID = 1
+};
+
+// 49028 - SPELL_DK_DANCING_RUNE_WEAPON
+class spell_dk_dancing_rune_weapon : public SpellScriptLoader
+{
+public:
+    spell_dk_dancing_rune_weapon() : SpellScriptLoader("spell_dk_dancing_rune_weapon") { }
+
+    class spell_dk_dancing_rune_weapon_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_dk_dancing_rune_weapon_AuraScript);
+
+        void HandleTarget(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            Unit* target = GetTarget();
+            if (!target)
+                return;
```

Comment:
according to Shauren, it's the other way round: only caster is ensured to never be null.